sources/api/netdog/src/networkd/config/netdev.rs

```diff
@@ -100,12 +108,214 @@ impl Display for ArpValidate {
+        }
+    }
+
+    /// Add bond minimum links
+    pub(crate) fn with_min_links(&mut self, min_links: usize) {
+        self.netdev.bond_mut().min_links = Some(min_links)
+    }
+
+    /// Add MIIMon configuration
+    pub(crate) fn with_miimon_config(&mut self, miimon: MiiMonitoringConfigV1) {
+        let bond = self.netdev.bond_mut();
+
+        bond.mii_mon_secs = Some(miimon.frequency);
+        bond.up_delay_sec = Some(miimon.updelay);
+        bond.down_delay_sec = Some(miimon.downdelay);
+    }
+
+    /// Add ARPMon configuration
+    pub(crate) fn with_arpmon_config(&mut self, arpmon: ArpMonitoringConfigV1) {
+        let bond = self.netdev.bond_mut();
+
+        // Legacy alert: wicked defaults to "any", keep that default here
+        // TODO: add a setting for this
+        bond.arp_all_targets = Some(ArpAllTargets::Any);
+        bond.arp_interval_secs = Some(arpmon.interval);
+        bond.arp_targets.extend(arpmon.targets);
+        bond.arp_validate = match arpmon.validate {
+            ArpValidateV1::Active => Some(ArpValidate::Active),
+            ArpValidateV1::All => Some(ArpValidate::All),
+            ArpValidateV1::Backup => Some(ArpValidate::Backup),
+            ArpValidateV1::None => Some(ArpValidate::r#None),
+        };
+    }
+}
+
+impl NetDevBuilder<Vlan> {
+    /// Create a new .netdev config for a VLAN
+    pub(crate) fn new_vlan(name: InterfaceName) -> Self {
+        let netdev = NetDevConfig {
+            netdev: Some(NetDevSection {
+                name: Some(name),
+                kind: Some(NetDevKind::Vlan),
+            }),
+            ..Default::default()
+        };
+
+        Self {
+            netdev,
+            spooky: PhantomData,
+        }
+    }
+
+    /// Add the VLAN's ID
+    pub(crate) fn with_vlan_id(&mut self, id: VlanId) {
+        self.netdev.vlan_mut().id = Some(id);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::bonding::BondMonitoringConfigV1;
+    use crate::networkd::config::tests::{builder_data, test_data, TestDevices};
+    use crate::networkd::devices::{NetworkDBond, NetworkDVlan};
+
+    fn netdev_path(name: String) -> PathBuf {
+        test_data().join("netdev").join(format!("{}.netdev", name))
+    }
+
+    fn netdev_from_bond(bond: NetworkDBond) -> NetDevConfig {
+        let mut netdev = NetDevBuilder::new_bond(bond.name.clone());
+        netdev.with_mode(bond.mode);
+        bond.min_links.map(|m| netdev.with_min_links(m));
+        match bond.monitoring_config {
+            BondMonitoringConfigV1::MiiMon(miimon) => netdev.with_miimon_config(miimon),
+            BondMonitoringConfigV1::ArpMon(arpmon) => netdev.with_arpmon_config(arpmon),
+        }
+        netdev.build()
+    }
+
+    fn netdev_from_vlan(vlan: NetworkDVlan) -> NetDevConfig {
+        let mut netdev = NetDevBuilder::new_vlan(vlan.name.clone());
+        netdev.with_vlan_id(vlan.id);
+        netdev.build()
+    }
```

Comment:
This logic will most likely go away once it's integrated into the device code.  It exists to enable unit testing now before that code is available.